pennylane/interfaces/batch/torch.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for adding the Autograd interface
+to a PennyLane Device class.
+"""
+# pylint: disable=too-many-arguments,protected-access
+import inspect
+
+import numpy as np
+import torch
+
+import pennylane as qml
+
+
+def _compute_vjp(dy, jacs, device=None):
+    vjps = []
+
+    for d, jac in zip(dy, jacs):
+        if isinstance(jac, np.ndarray):
+            jac = torch.from_numpy(jac)
+
+        jac = torch.as_tensor(jac, device=device)
+        vjp = qml.gradients.compute_vjp(d, jac)
+        vjps.extend(vjp)
+
+    return vjps
+
+
+class ExecuteTapes(torch.autograd.Function):
+    """The signature of this ``torch.autograd.Function`` is designed to
+    workaround Torch restrictions.
+
+    In particular, ``torch.autograd.Function``:
+
+    - Cannot accept keyword arguments. As a result, we pass a dictionary
+      as the first argument ``kwargs``. This dictionary **must** contain:
+
+      * ``"tapes"``: the quantum tapes to batch evaluate
+      * ``"device"``: the device to use to evaluate the tapes
+      * ``execute_fn``: the execution function to use on forward passes
+      * ``"gradient_fn"``: the gradient transform function to use
+        for backward passes
+      * ``gradient_kwargs``: gradient keyword arguments to pass to the
+        gradient function
+      * ``max_diff``: the maximum order of derivatives to support
+
+    Further, note that the ``parameters`` argument is dependent on the
+    ``tapes``; this Function should always be called
+    with the parameters extracted directly from the tapes as follows:
+
+    >>> parameters = []
+    >>> [parameters.extend(t.get_parameters()) for t in tapes])

```

Comment:
Extra paren at the end here; something looks funny with the syntax here, silly question, can a list comprehension be used like this to modify the list without actually assigning the result to anything?

Reply:
```suggestion
    >>> [parameters.extend(t.get_parameters()) for t in tapes]
```

> silly question, can a list comprehension be used like this to modify the list without actually assigning the result to anything?

Yep! basically this works purely due to the fact that `extend` is an in-place method, so it simply modifies `parameters` at every call.